jsonschema/src/keywords/custom_keyword_schema.rs

```diff
@@ -0,0 +1,101 @@
+use crate::compilation::context::CompilationContext;
+use crate::compilation::options::KeywordDefinition;
+use crate::error::no_error;
+use crate::keywords::CompilationResult;
+use crate::paths::{InstancePath, JSONPointer, PathChunk};
+use crate::validator::Validate;
+use crate::{ErrorIterator, JSONSchema, ValidationError};
+use serde_json::{Map, Value};
+use std::fmt::{Display, Formatter};
+
+pub(crate) struct CustomKeywordSchemaValidator {
+    schema_path: JSONPointer,
+    keyword_schema: Option<Value>,
+    json_schema: Option<JSONSchema>,
+}
+
+impl CustomKeywordSchemaValidator {
+    #[inline]
+    pub(crate) fn compile(
+        _: &Value,
+        schema_path: JSONPointer,
+        keyword_schema: Value,
+    ) -> CompilationResult {
+        let mut validator = CustomKeywordSchemaValidator {
+            schema_path,
+            keyword_schema: Some(keyword_schema),
+            json_schema: None,
```

Comment:
The general direction in other keywords is to avoid perf costs at runtime - if it could be detected at the compilation phase, then it is better to avoid it altogether.

If there is nothing in the child validator, then the parent one should not be compiled at all. For this reason, all the `compile` functions return `Option<CompilationResult<'a>>`.

I'd suggest rearranging the implementation a bit, so the `json_schema` field has the `JSONSchema` type and always has something to validate.